ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py

```diff
@@ -507,7 +512,70 @@ def __init__(self, config: WCA_PIPELINE_CONFIGURATION):
         super().__init__(config=config)
 
     def invoke(self, params: RoleGenerationParameters) -> RoleGenerationResponse:
-        raise NotImplementedError
+        request = params.request
```

Comment:
I believe this should only be implemented in `pipelines_saas.py` since we don't want to provide this feature yet with the onPrem offert. @manstis WDYT?

Reply:
@goneri @mabashian Yes, if this is only available for SaaS then that would be a better place for it. However please put a comment saying it should be moved back to the base WCA class when it becomes available for on-prem 😁 (otherwise we'll either have to remember or risk the duplication).

Reply:
(I was going to review this PR tomorrow)

Reply:
Thanks @manstis @goneri - I've moved this logic out to the saas pipeline